client/components/TemplateDraft.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect, useCallback } from "react";
+import { Row, Col } from "reactstrap";
+import Dante from "Dante2";
+
+import { saveTemplateDraft, getTemplateByID } from "../utils/apiWrapper";
+
+import "../public/styles/medium-draft.scss";
+import debounce from "lodash/debounce";
+
+import firebase from "firebase/app";
+import "firebase/storage";
+const firebaseConfig = {
+  apiKey: process.env.FIREBASE_APIKEY,
+  authDomain: process.env.AUTH_DOMAIN,
+  databaseURL: process.env.DATABASE_URL,
+  storageBucket: process.env.STORAGE_BUCKET
+};
+if (!firebase.apps.length) {
+  firebase.initializeApp(firebaseConfig);
+}
+const storageRef = firebase.storage().ref();
+
+export default function TemplateDraft({ id }) {
+  const [unsaved, setUnsaved] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  const [prevContent, setPrevContent] = useState(null);
+  const [editorContent, setEditorContent] = useState(null);
+
+  const saveInterval = 1000;
+  const debounceSave = (id_, json) => {
+    saveTemplateDraft(id_, json);
+    setUnsaved(false);
+  };
+  const saveCallback = useCallback(debounce(debounceSave, saveInterval), []);
```

Comment:
was this taken from brandon's stuff? i'm a bit confused as to why it needs to be a useCallback

Reply:
Yes, Ashwin took this part from Brandon's stuff in components/Draft.jsx. Looks like it doesn't need to be a useCallback though. Line 35 modified to:
`const saveCallback = debounce(debounceSave, saveInterval);`